src/frontend/src/binder/relation/join.rs

```diff
@@ -93,9 +103,58 @@ impl Binder {
                 }
                 bound_expr
             }
-            JoinConstraint::Using(_columns) => {
-                return Err(ErrorCode::NotImplemented("USING".into(), 1636.into()).into())
+            JoinConstraint::Using(columns) => {
+                let mut columns_iter = columns.into_iter();
+                let first_column = columns_iter.next().unwrap();
+                let mut binary_expr = Expr::BinaryOp {
+                    left: Box::new(Expr::CompoundIdentifier(vec![
+                        left_table.clone().unwrap(),
+                        first_column.clone(),
+                    ])),
+                    op: BinaryOperator::Eq,
+                    right: Box::new(Expr::CompoundIdentifier(vec![
+                        right_table.clone().unwrap(),
+                        first_column,
+                    ])),
+                };
+                for column in columns_iter {
+                    binary_expr = Expr::BinaryOp {
+                        left: Box::new(binary_expr),
+                        op: BinaryOperator::Eq,
+                        right: Box::new(Expr::BinaryOp {
+                            left: Box::new(Expr::CompoundIdentifier(vec![
+                                left_table.clone().unwrap(),
+                                column.clone(),
+                            ])),
+                            op: BinaryOperator::Eq,
+                            right: Box::new(Expr::CompoundIdentifier(vec![
+                                right_table.clone().unwrap(),
+                                column.clone(),
+                            ])),
+                        }),
+                    }
+                }
+                self.bind_expr(binary_expr)?
             }
         })
     }
 }
+
+fn get_table_name(table_factor: &TableFactor) -> Option<Ident> {
+    match table_factor {
+        TableFactor::Table {
+            name,
+            alias: _,
+            args: _,
+        } => Some(name.0[0].clone()),
+        TableFactor::Derived {
+            lateral: _,
+            subquery: _,
+            alias,
+        } => alias.as_ref().map(|table_alias| table_alias.name.clone()),
+        TableFactor::TableFunction { expr: _, alias } => {
```

Comment:
```suggestion
        TableFactor::Table {
            name,
            ..
        } => Some(name.0[0].clone()),
        TableFactor::Derived {
            alias,
            ..
        } => alias.as_ref().map(|table_alias| table_alias.name.clone()),
        TableFactor::TableFunction { alias, .. } => {
```
Could make the code cleaner 🥰